tutorials/io/plot_30_reading_fnirs_data.py

```diff
@@ -60,17 +61,135 @@
 If triggers are sent using the ``digaux`` port of the recording hardware, MNE
 will also read the ``digaux`` data and create annotations for any triggers.
 
-
-Storing of optode locations
-===========================
-
-NIRs devices consist of light sources and light detectors.
-A channel is formed by source-detector pairs.
-MNE stores the location of the channels, sources, and detectors.
-
-
-.. warning:: Information about device light wavelength is stored in
-             channel names. Manual modification of channel names is not
-             recommended.
-
+Loading legacy data in csv or tsv format
+========================================
+
+Many legacy fNIRS measurements are stored in csv and tsv formats.
+These formats are not officially supported in MNE as there is no
+standardisation of the file format -
+the naming and ordering of channels, the type and scaling of data, and
+specification of sensor positions varies between each vendor.
+Instead, we suggest that data is converted to the format approved by the
+Society for functional Near-Infrared Spectroscopy called
+`SNIRF <https://github.com/fNIRS/snirf>`_,
+they provide a number of tools to convert your legacy
+data to the SNIRF format.
+However, due to the prevalence of these legacy files we provide
+a template example of how you may read data in t/csv formats.
 """  # noqa:E501
+
+import numpy as np
+import pandas as pd
+import mne
+
+# sphinx_gallery_thumbnail_number = 2
+
+###############################################################################
+# First, we generate an example csv file which will then be loaded in to MNE.
+# This step would be skipped if you have actual data you wish to load.
+# We simulate 16 channels with 100 samples of data and save this to a file
+# called fnirs.csv.
+
+pd.DataFrame(np.random.normal(size=(16, 100))).to_csv("fnirs.csv")
+
+
+###############################################################################
+# Next, we will load the example csv file.
+# The metadata must be specified manually as the csv file does not contain
+# information about channel names, types, sample rate etc.
+
+data = pd.read_csv('fnirs.csv')
+
+# In MNE the naming of channels MUST follow this structure of
+# `S#_D# type` or `S#_D# wavelength`, where # is replaced
+# by the appropriate source and detector number, type is
+# either hbo or hbr, and wavelength is specified in nm.
+
+ch_names = ['D1_S1 hbo', 'D1_S1 hbr', 'D1_S2 hbo', 'D1_S2 hbr',
+            'D1_S3 hbo', 'D1_S3 hbr', 'D1_S4 hbo', 'D1_S4 hbr',
+            'D2_S5 hbo', 'D2_S5 hbr', 'D2_S6 hbo', 'D2_S6 hbr',
+            'D2_S7 hbo', 'D2_S7 hbr', 'D2_S8 hbo', 'D2_S8 hbr']
```

Comment:
THIS WILL NOT WORK. You must keep the naming structure as I described. All the downstream processing and MNE functions expect this.

Reply:
That's the issue indeed because of this:
```python
source, detector = ch.split(' ')[0].split('_')
```
Will fix it.

Reply:
Regex is probably better actually.

Reply:
sorted was a nice solution too